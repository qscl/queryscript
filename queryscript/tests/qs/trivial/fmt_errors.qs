select 1 as f"foo_{s}";
select f"{ss}" from (select 1 as "s"); -- should be the value 1
select f'{ss}'; -- should be the value s

select
    for item in slices {
        item AS f"metric_{item}"
    }
;
