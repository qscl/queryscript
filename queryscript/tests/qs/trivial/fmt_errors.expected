{
    "compile_errors": [
        (
            Some(
                0,
            ),
            NoSuchEntry {
                path: [
                    "s",
                ],
                backtrace: None,
            },
        ),
        (
            Some(
                1,
            ),
            NoSuchEntry {
                path: [
                    "ss",
                ],
                backtrace: None,
            },
        ),
        (
            Some(
                2,
            ),
            Unimplemented {
                what: "Format string values: f'{ss}'",
                backtrace: None,
                loc: Range(
                    "tests/qs/trivial/fmt_errors.qs",
                    Range {
                        start: Location {
                            line: 3,
                            column: 1,
                        },
                        end: Location {
                            line: 3,
                            column: 14,
                        },
                    },
                ),
            },
        ),
        (
            None,
            NoSuchEntry {
                path: [
                    "slices",
                ],
                backtrace: None,
            },
        ),
    ],
    "decls": {},
    "queries": [
        Err(
            StringError {
                what: "Unknown type cannot exist at runtime (?slot?)",
                backtrace: None,
            },
        ),
    ],
}